snapcraft/plugins/v2/_ros.py

```diff
@@ -66,8 +66,17 @@ def _get_workspace_activation_commands(self) -> List[str]:
         return [
             'state="$(set +o)"',
             "set +u",
+            'if [ -f "${SNAPCRAFT_PART_INSTALL}"/opt/ros/$ROS_DISTRO/setup.sh ]; then',
+            "set -- --local",
+            "_CATKIN_SETUP_DIR={path} . {path}/setup.sh".format(
+                path='"${SNAPCRAFT_PART_INSTALL}"/opt/ros/$ROS_DISTRO'
+            ),
+            "set -- --local --extend",
+            "else",
+            "set -- --local",
+            "fi",
             ". /opt/ros/$ROS_DISTRO/setup.sh",
-            'eval "$(state)"',
+            'eval "${state}"',
```

Comment:
@cjp256 we seem to go back and forth on this. How about we add a code comment :-)

Reply:
Here `state` is a variable, not a function,

```bash
$ state="echo ok"
$ $(state)
Command 'state' not found, did you mean: ...
$ ${state}
ok
```

Hence this fix.
A similar snippet can be found in the [Python plugin](https://github.com/snapcore/snapcraft/blob/405c576553d22be9727f681f8cda88b220b39f30/snapcraft/plugins/v2/python.py#L180).

Reply:
@sergiusens I imagine you're thinking of https://github.com/snapcore/snapcraft/commit/f0b2f95fce9b5a4360954784b67046fcd2409103

@artivis' fix is correct, thanks Jeremy!